src/lib/Day2/IntcodeRunner.cs

```diff
@@ -44,36 +55,142 @@ public int[] Execute()
         public int[] Execute(int noun, int verb)
         {
             var output = (int[])this.instructions.Clone();
+            var outputIndex = -1;
             output[1] = noun;
             output[2] = verb;
 
-            int index = 0;
-            var opcode = (Opcode)output[index];
+            int index = 0, numParams = 0;            
+            var (opcode, operModes) = ParseOpcode(output[index]);
+
```

Comment:
Initialize numParams to default (most common) value here (2), then update in switch statement if needed.